### PR TITLE
Documentation for 0.6

### PR DESCRIPTION
**Pull request by cerveada**
This is a draft of Spline 0.6 documentation.

### CODE REVIEW

File: tmp-0.6/gateway0.6.MD [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+First we will download ArangoDB in docker and run it:
+```shell
+docker run -p 8529:8529 -e ARANGO_NO_AUTH=1 arangodb/arangodb:3.7.10
+```
+Once your database is running you should be able to see ArangoDB web interface at http://localhost:8529.
+
+Then we have to initialize Spline database. Create collections, indices, views etc.
+```shell
+docker run absaoss/spline-admin:0.6.0 ??? 
+// TODO
```
Comment (cerveada) [inline tmp-0.6/gateway0.6.MD]:
```suggestion
docker run -it --rm absaoss/spline-admin:0.6.0 db-init arangodb://localhost/spline
```
@wajda Not sure how to call this via docker, would this work?

File: tmp-0.6/gateway0.6.MD [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+`spline.database.logFullQueryOnError` (default false)
+- Will log the full arango query when se t to true.
+
+### Consumer Properties
+`spline.adaptive_timeout.min` (default 3s)
+- TODO
+
+`spline.adaptive_timeout.duration_factor` (default 1.5)
+- TODO
+
```
Comment (cerveada) [inline tmp-0.6/gateway0.6.MD]: Should this be in documentation? According to context.xml it's not used.

File: tmp-0.6/gateway0.6.MD [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+... and other useful URLs:
+  - Running server version information: [/about/version](http://localhost:8080/about/version)
+  - Producer API Swagger documentation: [/docs/producer.html](http://localhost:8080/docs/producer.html) 
+  - Consumer API Swagger documentation: [/docs/consumer.html](http://localhost:8080/docs/consumer.html) 
+
+## Configuration
+Spline is looking for configuration properties in those sources:
+- JVM system properties
+- `spline.properties` file on classpath
+
```
Comment (wajda) [inline tmp-0.6/gateway0.6.MD]:
```suggestion
Spline is looking for configuration properties in the following sources (order by precedence):
- Naming and directory context (via JNDI) if available
- JVM system properties
- System environment variables

Please adhere to the naming convention of each configuration source type. E.g. the property `foo.barBaz` would be looked up as `foo.barBaz` in the JVM options, as `foo/barBaz` in the JNDI, and as `FOO_BAR_BAZ` in the environment variables.
For each source the 

```

File: tmp-0.6/gateway0.6.MD (hunk 2 above)
Comment (wajda) [reply tmp-0.6/gateway0.6.MD]: No, it's not used. Let's delete it from the doc.

File: tmp-0.6/gateway0.6.MD [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+### Persistance Properties
+`spline.database.connectionUrl`
+- Url of Arango database.
+
+`spline.database.logFullQueryOnError` (default false)
+- Will log the full arango query when se t to true.
```
Comment (wajda) [inline tmp-0.6/gateway0.6.MD]:
```suggestion
- Will log the full AQL query when set to true.
```

File: tmp-0.6/gateway0.6.MD [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+### Consumer Properties
+`spline.adaptive_timeout.min` (default 3s)
+- TODO
+
+`spline.adaptive_timeout.duration_factor` (default 1.5)
+- TODO
+
+### Kafka Gateway Properties
+`spline.kafka.topic`
+- Topic from which the gateway will consume messages.
```
Comment (wajda) [inline tmp-0.6/gateway0.6.MD]:
```suggestion
- A topic from which the gateway will consume messages.
```

File: tmp-0.6/spark-agent0.6.MD [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+|------------|:----------------------------:|:----------:|
+|**Spark 2.2** | (no SQL; no codeless init) | &mdash;    |
+|**Spark 2.3** | (no Delta support)         | &mdash;    |
+|**Spark 2.4** | Yes                        | Yes        |
+|**Spark 3.0** | &mdash;                    | Yes        |
+
+
+## Artifacts
+- `agent-core_{{SPARK_VERSION}}` is a classic maven library that you can use with any compatible Spark version. This is required if you want to have acces to agent classes or use **programatic initialization**. 
+- `spark-{{SCALA_VERSION}}-spline-agent-bundle_{{SPARK_VERSION}}` is a fat jar containing all dependencies inside omitting the ones already provided by Spark. You can use the bundle in Spark via `--packages` flag. If you also activate the agent via **codeless initialization** your own code doesn't have to have any dependency on the agent it will be used directly by Spark.
```
Comment (wajda) [inline tmp-0.6/spark-agent0.6.MD]: Looks like you confused Scala version with Spark version. Can you check the text and fix where needed? (I didn't seek for all occurrences)

File: tmp-0.6/spark-agent0.6.MD [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+
+```bash
+pyspark \
+  --packages za.co.absa.spline.agent.spark:spark-2.4-spline-agent-bundle_2.12:0.6.0 \
+  --conf "spark.sql.queryExecutionListeners=za.co.absa.spline.harvester.listener.SplineQueryExecutionListener" \
+  --conf "spline.lineageDispatcher.http.producer.url=http://localhost:9090/producer"
+```
+
+## Configuration
+Agent is looking for configuration properties in those sources:
```
Comment (wajda) [inline tmp-0.6/spark-agent0.6.MD]:
```suggestion
The agent looks for configuration properties in the following sources:
```

File: tmp-0.6/spark-agent0.6.MD [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+- Hadoop configuration (core-site.xml)
+- Spark configuration
+- JVM system properties
+- `spline.properties` file on classpath
+
+File in agent `spline.default.properties` contains default values for properties as well as some additional properties docuemntation.
```
Comment (wajda) [inline tmp-0.6/spark-agent0.6.MD]:
```suggestion
The file `spline.default.properties` (in the agent source code) contains default values for properties as well as some additional documentation.
```

File: tmp-0.6/spark-agent0.6.MD [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+It's good idea to look in the file to see what properties are available.
+
+### Properties
+
+`spline.mode`
+- `DISABLED` Lineage tracking is completely disabled and Spline is unhooked from Spark.
+- `REQUIRED` If Spline fails to initialize itself (e.g. wrong configuration, no db connection etc) the Spark application aborts with an error.
```
Comment (wajda) [inline tmp-0.6/spark-agent0.6.MD]:
```suggestion
- `REQUIRED` If Spline fails to initialize itself (e.g. wrong configuration, no db connection etc) the Spark application aborts with an error. (**Note**: that only concerns Spline initialization routine. If the error happens during lineage capturing, or in the Spline dispatcher, then the target Spark job will have already been finished by that time, and the resulted data will have been persisted, regardless of the `spline.mode` settings. The Spline agent doesn't do any automated rollbacks).
```

File: tmp-0.6/spark-agent0.6.MD [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+
+`spline.IWDStrategy.default.onMissingMetrics` - behaviour when info whether the data was actually written is missing
+- `CAPTURE_LINEAGE`
+- `IGNORE_LINEAGE` (default)
+
+`spline.postprocessingFilter.classNames`
+ - Optional comma separated list of user lineage filters. More info in **Post Processing Lineage Filters** chapter.
+
+## Lineage Dispatchers
+Lineage Dispatchers is a class that is usually (but not always) responsible for sending the converted lineage to gateway. By default `HttpLineageDispatcher` is used.
```
Comment (wajda) [inline tmp-0.6/spark-agent0.6.MD]:
```suggestion
The `LineageDispatcher` trait is responsible for sending out the captured lineage information. By default, the `HttpLineageDispatcher` is used, that sends the lineage data to the Spline REST endpoint (see Spline Producer API).
```

File: tmp-0.6/spark-agent0.6.MD [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+## Post Processing Lineage Filters
+Filters can be used to enrich the lineage with your own custom data or to remove unwanted data like passwords. All filters are aplied after the Spark plan is converted to Spline DTOs, but before the dispatcher is called.
+
+The filter must implement `LineageFilter` class and have a constructor with a single parameter of type `org.apache.commons.configuration.Configuration`. For an example you can take a look at `DataSourceUriLineageFilter`. You can use the Configuration instance to acces configuration params and make the filter configurable that way. `LineageFilter`'s methods will provide you not only with lineage DTOs, but also wtih `HarvestingContext` object containing the original Spark plan.
+
+To use the filter you must set it via a property. For example:
+```
+spline.postprocessingFilter.classNames=za.co.absa.spline.harvester.postprocessing.DataSourceUriLineageFilter
+```
+
```
Comment (wajda) [inline tmp-0.6/spark-agent0.6.MD]:
``````suggestion
The procedure how filters are registered and configured is similar to the `LineageDispatcher` registration and configuration procedure.
A custom filter class must implement `za.co.absa.spline.harvester.postprocessing.PostProcessingFilter` trait and declare a constructor with a single parameter of type `org.apache.commons.configuration.Configuration`.
Then register and configure it like this: 
```properties
spline.postProcessingFilter.className=my-filter
spline.postProcessingFilter.my-filter.className=my.awesome.CustomFilter
spline.postProcessingFilter.my-filter.prop1=value1
spline.postProcessingFilter.my-filter.prop2=value2
```

Use pre-registered `CompositePostProcessingFilter` to chain up multiple filters:
```properties
spline.postProcessingFilter=composite
spline.postProcessingFilter.composite.filters=myFilter1,myFilter2
```

(see `spline.default.properties` for details and examples)
``````

File: tmp-0.6/spark-agent0.6.MD (hunk 6 above)
Comment (cerveada) [reply tmp-0.6/spark-agent0.6.MD]: I am thinking about removing the Scala version completely from the name. The Scala version at the end is a standard used by everyone. Specifically mentioning it here can make it more confusing.

**Pull request merged**